SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Utils/ErrorLogHelper.cs

```diff
@@ -219,6 +225,25 @@ public virtual ManagedErrorLog InstanceReadErrorLogFile(File file)
             return null;
         }
 
+        /// <summary>
+        /// Creates the error storage directory if it does not exist.
+        /// </summary>
+        /// <returns>The error storage directory.</returns>
+        public virtual Directory InstanceGetErrorStorageDirectory()
+        {
+            if (!_crashesDirectory.Exists())
+            {
+                lock (LockObject)
+                {
+                    if (!_crashesDirectory.Exists())
```

Comment:
We don't need to check twice, actually Create is already checking, also no need to lock.
I think this was already discussed with @achocron  on another task when we looked at the code of Create.

Reply:
Yeah, this is a strange pattern.  I'd be surprised if we needed to check twice.  I would think that the outermost check could be removed.

Reply:
The double-checking pattern saves us from locking unnecessarily so I see the motivation there. But as @guperrot said, we probably don't actually need to check even once. It's fine to just call Create without checking.

Reply:
Thanks for the input guys :) I learned that pattern from @achocron , as he said :)